libkbfs/backpressure_disk_limiter_test.go

```diff
@@ -15,439 +16,564 @@ import (
+	default:
+		panic(fmt.Sprintf("unknown test type %s", testType))
+	}
 
 	log := logger.NewTestLogger(t)
 	bdl, err := newBackpressureDiskLimiterWithFunctions(
-		log, 0.1, 0.9, 0.25, 10*blockSize, 8*time.Second, delayFn,
-		func() (int64, error) {
-			return math.MaxInt64, nil
+		log, 0.1, 0.9, 0.25, byteLimit, fileLimit,
+		8*time.Second, delayFn,
+		func() (int64, int64, error) {
+			return math.MaxInt64, math.MaxInt64, nil
 		})
 	require.NoError(t, err)
 
-	journalBytes, freeBytes, byteSemaphoreMax :=
-		bdl.getLockedVarsForTest()
-	require.Equal(t, int64(0), journalBytes)
-	require.Equal(t, int64(math.MaxInt64), freeBytes)
-	require.Equal(t, int64(100), byteSemaphoreMax)
-	require.Equal(t, int64(100), bdl.byteSemaphore.Count())
+	byteSnapshot, fileSnapshot := bdl.getSnapshotsForTest()
+	require.Equal(t, bdlSnapshot{
+		used:  0,
+		free:  math.MaxInt64,
+		max:   byteLimit,
+		count: byteLimit,
+	}, byteSnapshot)
+	require.Equal(t, bdlSnapshot{
+		used:  0,
+		free:  math.MaxInt64,
+		max:   fileLimit,
+		count: fileLimit,
+	}, fileSnapshot)
 
 	ctx := context.Background()
 
-	var bytesPut int
-
-	checkCountersAfterBeforeBlockPut := func() {
-		journalBytes, freeBytes, byteSemaphoreMax =
-			bdl.getLockedVarsForTest()
-		require.Equal(t, int64(bytesPut), journalBytes)
-		require.Equal(t, int64(math.MaxInt64), freeBytes)
-		require.Equal(t, int64(100), byteSemaphoreMax)
-		require.Equal(t, int64(100-bytesPut-blockSize),
-			bdl.byteSemaphore.Count())
+	var bytesPut, filesPut int64
+
+	checkCountersAfterBeforeBlockPut := func(i int) {
+		byteSnapshot, fileSnapshot := bdl.getSnapshotsForTest()
+		require.Equal(t, bdlSnapshot{
+			used:  bytesPut,
+			free:  math.MaxInt64,
+			max:   byteLimit,
+			count: byteLimit - bytesPut - blockBytes,
+		}, byteSnapshot, "i=%d", i)
+		require.Equal(t, bdlSnapshot{
+			used:  filesPut,
+			free:  math.MaxInt64,
+			max:   fileLimit,
+			count: fileLimit - filesPut - blockFiles,
+		}, fileSnapshot, "i=%d", i)
 	}
 
-	checkCountersAfterBlockPut := func() {
-		journalBytes, freeBytes, byteSemaphoreMax =
-			bdl.getLockedVarsForTest()
-		require.Equal(t, int64(bytesPut), journalBytes)
-		require.Equal(t, int64(math.MaxInt64), freeBytes)
-		require.Equal(t, int64(100), byteSemaphoreMax)
-		require.Equal(t, int64(100-bytesPut),
-			bdl.byteSemaphore.Count())
+	checkCountersAfterBlockPut := func(i int) {
+		byteSnapshot, fileSnapshot := bdl.getSnapshotsForTest()
+		require.Equal(t, bdlSnapshot{
+			used:  bytesPut,
+			free:  math.MaxInt64,
+			max:   byteLimit,
+			count: byteLimit - bytesPut,
+		}, byteSnapshot, "i=%d", i)
+		require.Equal(t, bdlSnapshot{
+			used:  filesPut,
+			free:  math.MaxInt64,
+			max:   fileLimit,
+			count: fileLimit - filesPut,
+		}, fileSnapshot, "i=%d", i)
 	}
 
 	// The first two puts shouldn't encounter any backpressure...
 
 	for i := 0; i < 2; i++ {
-		_, _, err = bdl.beforeBlockPut(ctx, blockSize, 0)
+		_, _, err = bdl.beforeBlockPut(ctx, blockBytes, blockFiles)
```

Comment:
Should probably be checking these values to avoid the bug I noted above.  (Though it could be done in a simpler test, as well.)